### PR TITLE
normalize to using airflow Variable.get for env specific values

### LINKED ISSUE #80: Support for Sinopia Stage and Production
Currently we only been using Sinopia Development environment. We need to support Sinopia Stage first and then Production environments.

### PR DESCRIPTION
**Pull request by jmartin-sul**
including environment name.

* modify tests to mock variable retrieval or to provide env vars for airflow's Variable.get to use
* provide some default values for dev in docker-compose.yml
* assume that the app code for an instance operates on one and only one environment at a time.  i.e. there no "foo_dev" and "foo_stage" vars, because the instance is either entirely for stage or entirely for dev.

TODO:
- [x] a bit of manual testing on my local instance
- [x]  document which variables and connections need to be renamed in the wiki (`aws_ses_dev` -> `aws_ses_connection`, `aws_sqs_dev` -> `aws_sqs_connection`, `stanford_symphony_dev_login` -> `stanford_symphony_login`, `stanford_symphony_dev_password` `stanford_symphony_password`, ???).  will need to update the wiki.
- [ ] more helm chart updating?

closes #80

### CODE REVIEW

**Comment by jmartin-sul** [comment]
* manual testing:  i was able to do the same thing i'd been doing to test the failure task, which is queue a message using my local sinopia api instance, pick it up in my local airflow instance, and have processing fail at the message parse step because my airflow instance can't retrieve the resource from my sinopia_api instance.  that behaved as expected -- the emails and honeybadger notifications were sent.  but that's not a very thorough test.  happy to pair or take advice on testing the other steps in the DAG, if more testing is warranted before merging.
* i think the documentation changes, if this is merged as-is or in a similar state, would be:
  * https://github.com/LD4P/ils-middleware/wiki/AWS-SQS#current-sqs-queues-and-adding-as-airflow-variables -- `SQS_DEV` and `SQS_STAGE` to `SQS_URL` (with a note that the docker-compose defaults give a value via `AIRFLOW_VAR_SQS_URL`)
  * https://github.com/LD4P/ils-middleware/wiki/AWS-SQS#setting-up-aws-airflow-connection -- (in screenshot?) `aws_sqs_dev` to `aws_sqs_connection`
  * https://github.com/LD4P/ils-middleware/wiki/Symphony-ILS-in-Airflow#airflow-variables -- `stanford_symphony_dev_login` -> `stanford_symphony_login`, `stanford_symphony_dev_password` -> `stanford_symphony_password`
  * related, still need to add SES config info, similar to SQS config info, per https://github.com/LD4P/ils-middleware/pull/83#issuecomment-956370891
* i'm not sure about the helm chart thing; @aaron-collier or @jgreben or @jermnelson?

anyway, might need further tweaking, but ready for review

**Review by jermnelson — APPROVED** [review]
I like the overall direction in pulling from Airflow Variables the env specific values. Since we're using the ECS to host our Airflow instance (for now instead of K8s), the `charts/values.yaml` file could be removed (and I believe we are using the Helm chart in TF and not this file).

File: docker-compose.yaml [hunk 1]
```diff
@@ -60,14 +60,15 @@ x-airflow-common:
     AIRFLOW__CORE__DAGS_ARE_PAUSED_AT_CREATION: 'true'
     AIRFLOW__CORE__LOAD_EXAMPLES: 'false'
     AIRFLOW__API__AUTH_BACKEND: 'airflow.api.auth.backend.basic_auth'
-    AIRFLOW_VAR_SQS_DEV: 'https://sqs.us-west-2.amazonaws.com/418214828013/stanford-ils'
-    DEPLOYMENT_ENV: ${DEPLOYMENT_ENV:-dev}
+    AIRFLOW_VAR_MARC_S3_BUCKET: 'sinopia-marc-development'
```
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply docker-compose.yaml:63]: > would setting these variables through the Airflow UI override these values when running `docker compose` locally?

not sure, but i'll test that!  my assumption was that they would, but looking back through the docs, i don't see any indication of precedence one way or the other.

> would some of these values be set in Terraform (see this PR sul-dlss/terraform-aws#913) or would the better location be in Vault?

my guess is that e.g. login type stuff (password, or user/pass) would be best stored in vault... and everything else through airflow UI/config vars, or env vars from the container env> (e.g. `docker compose` locally or terraform application for shared envs like stage/prod)

[the remark this answers is not in the document]
Comment (jmartin-sul) [reply docker-compose.yaml:63]: naive question (that i didn't see an answer to in the docs): will values stored in vault be retrievable via `Variable.get`?  or is there a different way of getting vault values?

[the remark this answers is not in the document]
Comment (jermnelson) [reply docker-compose.yaml:63]: It is my understanding these values are retrievable via `Variable.get`. I need to test to confirm.

**Pull request merged**